homeassistant/components/openhome/__init__.py

```diff
@@ -1 +1,51 @@
 """The openhome component."""
+
+import asyncio
+import logging
+
+import aiohttp
+from async_upnp_client.client import UpnpError
+from openhomedevice.device import Device
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOST, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+
+from .const import DOMAIN
+from .media_player import OpenhomeDevice
```

Comment:
You cannot import entities from platforms here. Your API object should live separately from the entity, and be instantiated here in `__init__.py`, it should be wrapped by an entity in `media_player.py`.

Reply:
Thanks. The `__init__` function now creates just the `openhomedevice.device` and adds that to `hass.data`.
The `media_player.py` entity now creates itself from that device. 

I've removed the `async_setup_platform` from `media_player.py` as a tidy up so it's no longer called by the discovery component if enabled.